sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -362,24 +362,14 @@ public void addEvent(io.opentelemetry.trace.Event event, long timestamp) {
     addTimedEvent(TimedEvent.create(timestamp, event));
   }
 
-  static Map<String, AttributeValue> copyAndLimitAttributes(
-      Map<String, AttributeValue> attributes, int limit) {
-    if (attributes.isEmpty()) {
-      return Collections.emptyMap();
-    }
-
-    if (attributes.size() <= limit) {
-      return Collections.unmodifiableMap(new HashMap<>(attributes));
-    }
-
-    Map<String, AttributeValue> temp = new HashMap<>();
-    for (Map.Entry<String, AttributeValue> entry : attributes.entrySet()) {
-      if (temp.size() < limit) {
-        temp.put(entry.getKey(), entry.getValue());
-      }
+  static Attributes copyAndLimitAttributes(final Attributes attributes, final int limit) {
+    if (attributes.isEmpty() || attributes.size() <= limit) {
+      return attributes;
     }
 
-    return Collections.unmodifiableMap(temp);
+    Attributes.Builder temp = Attributes.newBuilder();
```

Comment:
Would usually use something like `result`, not used to temp being as important as this one :)

Reply:
agreed. will pick this up in the next round of changes.